lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1067,6 +1087,16 @@ class NGPHistoryParser {
   }
 
   buildSettingsRecords(events) {
+    // Determine if the pump is in 'Auto Mode' by looking at the most recent history event
+    // and checking the Closed Loop state.
+    // If the pump is not in Closed Loop, we keep the active basal pattern from the status message.
+    const lastEvent = this.events[this.events.length - 1];
+    if (lastEvent.isClosedLoopActive) {
```

Comment:
Can we be confident that the last event in the pump history will always have the "closed loop active" flag? What if it's an alarm?

Reply:
Yes we can. _Every_ event contains this flag. They overloaded the source ID field, so it’s even backward compatible.